### PR TITLE
Add CrossBrowserTesting Service

### PR DESCRIPTION
**Pull request by daphnemcrossbrowser**
## Proposed changes

[//]: # Adds a CrossBrowserTesting service to WDIO that manages local tunnel and job metadata for CrossBrowserTesting users.

## Types of changes

[//]: # (What types of changes does your code introduce to WebdriverIO?)
[//]: # (_Put an `x` in the boxes that apply_)

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [x ] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/master/CONTRIBUTING.md) doc
- [x ] I have added tests that prove my fix is effective or that my feature works
- [ x] I have added necessary documentation (if appropriate)

## Further comments

[//]: # (If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...)

### CODE REVIEW

File: packages/wdio-crossbrowsertesting-service/src/launcher.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        this.cbtOpts = Object.assign({
+            username: config.user,
+            authkey: config.key
+        }, config.cbtOpts)
+        this.tunnel = cbt
+        return new Promise(function(resolve, reject){
+            this.tunnel.start({'username': config.user,'authkey': config.key},function(err){
+                if(!err){  
+                    return resolve()
+                }else{
```
Comment (mgrybyk) [inline packages/wdio-crossbrowsertesting-service/src/launcher.js]: I think there are some formatting issues here? I mean in whole file.

**Comment by mgrybyk** [comment]
Sorry to say I'm not familiar with CBT, so can't help much on validation of this one

File: packages/wdio-crossbrowsertesting-service/src/launcher.js (hunk 1 above)
Comment (daphnemcrossbrowser) [reply packages/wdio-crossbrowsertesting-service/src/launcher.js]: I'm currently working to fix these issues and add unit test.

File: packages/wdio-crossbrowsertesting-service/tests/launcher.test.js [hunk 2]
```diff
@@ -0,0 +1 @@
+
```
Comment (christian-bromann) [inline packages/wdio-crossbrowsertesting-service/tests/launcher.test.js]: You have to provide unit test coverage otherwise this service can't be merged

Comment (daphnemcrossbrowser) [reply packages/wdio-crossbrowsertesting-service/tests/launcher.test.js]: I'm currently working to add unit test.

**Comment by christian-bromann** [comment]
> 98.5% (-0.09%) compared to 99526a6

Can we bump the coverage a bit more? It seems like there is still some opportunity.

```
 wdio-crossbrowsertesting-service/src       |    76.67 |    70.31 |    82.61 |    77.01 |                   |
  launcher.js                               |    71.43 |    66.67 |       80 |       75 |          19,20,29 |
  service.js                                |    77.63 |    70.69 |    83.33 |    77.33 |... 84,186,187,214 |
```

File: packages/wdio-crossbrowsertesting-service/src/launcher.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
+var cbt = require('cbt_tunnels')
```
Comment (mgrybyk) [inline packages/wdio-crossbrowsertesting-service/src/launcher.js]: Can we use here import instead or at least const? 
Do we need to have cbt tunnels added as dependency?

Comment (daphnemcrossbrowser) [reply packages/wdio-crossbrowsertesting-service/src/launcher.js]: cbt_tunnels is added as a dependency in the package.json for the service.

**Comment by christian-bromann** [comment]
@daphnemcrossbrowser mind providing us with an account to test changes in here? Do you guys have something like an open source account type? Would be nice to have this and not having to create a free trial account every time.

**Comment by mgrybyk** [comment]
Some conflicts have to be resolved and test coverage to be increased, other than that looks good for me too.
I'm not familiar with CBT and haven't tested it locally.

**Comment by christian-bromann** [comment]
Resolved the conflict. Code coverage is fine. I the path coverage is not required, as long as the overall project coverage increases it's all good.

**Comment by christian-bromann** [comment]
> **codecov/project ** — 98.93% (-0.12%) compared to 3afbdfe

I am sorry @daphnemcrossbrowser , after I resolved the conflict in the config the coverage dropped again. Can you take a look?

**Comment by daphnemcrossbrowser** [comment]
@christian-bromann I'll take a look, and an open source account can be created here: https://crossbrowsertesting.com/open-source

**Comment by baruchvlz** [comment]
Please add this to the list of supported services in the CLI config as well as the CLI doc.

**Pull request merged**